client/components/term-tree-selector/terms.jsx

```diff
@@ -362,7 +361,7 @@ const TermTreeSelectorList = React.createClass( {
 
 		return (
 			<div ref={ this.setSelectorRef } className={ classes }>
-				{ this.state.requestedPages.map( ( page ) => (
+				{ siteId && this.state.requestedPages.map( ( page ) => (
```

Comment:
I think we'll need to update `componentWillReceiveProps` to reset the default state when `siteId` changes.